moto/elbv2/models.py

```diff
@@ -1308,41 +1319,23 @@ def modify_listener(
 
             # HTTPS checks
             if protocol == "HTTPS":
-                # HTTPS
-
-                # Might already be HTTPS so may not provide certs
-                if certificates is None and listener.protocol != "HTTPS":
-                    raise RESTError(
-                        "InvalidConfigurationRequest",
-                        "Certificates must be provided for HTTPS",
-                    )
-
                 # Check certificates exist
                 if certificates is not None:
-                    default_cert = None
-                    all_certs = set()  # for SNI
-                    for cert in certificates:
-                        if cert["is_default"] == "true":
-                            default_cert = cert["certificate_arn"]
-                        try:
-                            self.acm_backend.get_certificate(cert["certificate_arn"])
-                        except Exception:
-                            raise RESTError(
-                                "CertificateNotFound",
-                                "Certificate {0} not found".format(
-                                    cert["certificate_arn"]
-                                ),
-                            )
-
-                        all_certs.add(cert["certificate_arn"])
-
-                    if default_cert is None:
+                    default_cert = certificates[0]
```

Comment:
🐛  If `certificates` is an empty list, `certificates[0]` will raise an `IndexError`. I think the condition should match the one in `convert_and_validate_certificates`, i.e. `if certificates:`

Reply:
100% makes sense

Reply:
Good catch: 1b194f5

Reply:
Which also means this bug always existed